components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  //
+  // workspace_id +return NOT_FOUND Workspace does not exist
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  //
+  // workspace_id +return NOT_FOUND Workspace does not exist
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserEnvironmentVariable {
+  string id = 1;
+  string name = 2;
+  string value = 3;
+  string repository_pattern = 4;
+}
+
+message ListEnvironmentVariablesRequest { optional string workspace_id = 1; }
+
+message ListEnvironmentVariablesResponse {
+  repeated UserEnvironmentVariable environment_variables = 1;
+}
+
+message UpdateEnvironmentVariableRequest {
+  optional string envvar_name = 1;
+  optional string envvar_value = 2;
+  optional string envvar_repository_pattern = 3;
+}
+
+message UpdateEnvironmentVariableResponse {
+  UserEnvironmentVariable environment_variable = 1;
+}
+
+message DeleteEnvironmentVariableRequest {
+  string envvar_name = 1;
```

Comment:
it should have only `env_var_id`

Reply:
I did that at first but then I would need to change dashboard logic too and thought maybe better to improve api as a follow up or should it be done at the same time?

Reply:
How big will be a change? We could do in 2 steps:
- change JSON-RPC and dashboard APIs
- then add shim

I have the same trouble now with https://github.com/gitpod-io/gitpod/pull/19062